packages/devtools/lib/src/memory/memory_data_view.dart

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:vm_service_lib/vm_service_lib.dart';
+
+import '../memory/memory_controller.dart';
+import '../ui/custom.dart';
+import '../ui/elements.dart';
+
+typedef MemoryDescriber = Future<String> Function(BoundField variable);
+
+class MemoryDataView implements CoreElementView {
+  MemoryDataView(this._memoryController, MemoryDescriber variableDescriber) {
+    _items = SelectableTree<BoundField>()
+      ..flex()
+      ..clazz('menu-item-bottom-border')
+      ..clazz('debugger-items-list')
+      ..clazz('memory-inspector-items-list');
+
+    _items.setChildProvider(new MemoryDataChildProvider(_memoryController));
+
+    _items.setRenderer((BoundField field) {
+      final String name = field.decl.name;
+      final dynamic value = field.value;
+
+      String valueStr;
+
+      if (value is InstanceRef) {
+        if (value.valueAsString == null) {
+          valueStr = value.classRef.name;
+        } else {
+          valueStr = value.valueAsString;
+          if (value.valueAsStringIsTruncated) {
+            valueStr += '...';
+          }
+          if (value.kind == InstanceKind.kString) {
+            valueStr = "'$valueStr'";
+          }
+        }
+
+        if (value.kind == InstanceKind.kList) {
+          valueStr = '[${value.length}] $valueStr';
+        } else if (value.kind == InstanceKind.kMap) {
+          valueStr = '{ ${value.length} } $valueStr';
+        } else if (value.kind != null && value.kind.endsWith('List')) {
+          // Uint8List, Uint16List, ...
+          valueStr = '[${value.length}] $valueStr';
+        }
+      } else if (value is Sentinel) {
+        valueStr = value.valueAsString;
```

Comment:
what does this display? 
We want something user friendly to clarify the object probably gced.

Reply:
I'll add a TODO an investigate (debugger uses similar mechanism).

Reply:
I think it's sentinel but I'll double check too.